platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ImportMigrationManagerImpl.java

```diff
@@ -141,8 +143,10 @@ public ImportMigrationManagerImpl(
       this.currentProductBranding = currentProductBranding;
       this.currentProductVersion = currentProductVersion;
       // process migratables' metadata
-      JsonUtils.getMapFrom(metadata, MigrationContextImpl.METADATA_MIGRATABLES)
+      JsonUtils.getMapFrom(metadata, MigrationContextImpl.METADATA_MIGRATABLES, false)
           .forEach((id, o) -> getContextFor(id).processMetadata(JsonUtils.convertToMap(o)));
+      customSystemProperties = this.getImportedCustomSystemProperties(metadata);
```

Comment:
❗️ This line needs to be conditional based on the version. For older versions, we do what you are attempting to do in `getImportedCustomSystemProperties(). For recent version, they should just load the system.properties entry you put in the metadata.

Reply:
In `getImportedCustomSystemProperties` I actually try to do that. If the version is equal to `SYSTEM_PROPERTIES_MISSING_VERSION`, it tries to get the props, otherwise, the return statement at the bottom says to just get the properties from the metadata.